cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/aws-orchestrator/AwsOrchestrator.service.tsx

```diff
@@ -87,4 +89,50 @@ export class AwsOrchestrator {
             setErrorObject(addErrorInfo(errorObject, e, AWS_FIELDS.NODE_TYPE));
         }
     }
+
+    static initAvailabilityZones(props: AwsOrchestratorProps) {
+        const { awsDispatch, setErrorObject, errorObject } = props;
+        DefaultOrchestrator.initResources<AWSKeyPair>({
+            resourceName: AWS_FIELDS.AVAILABILITY_ZONES,
+            dispatch: awsDispatch,
+            errorObject,
+            setErrorObject,
+            fetcher: () => AwsService.getAwsAvailabilityZones(),
+        });
+    }
+
+    static async initNodeTypes4AZs(props: AwsOrchestratorProps, nodeProfile: string) {
+        const { awsState, awsDispatch, setErrorObject, errorObject } = props;
+        const azList = awsState[STORE_SECTION_RESOURCES][AWS_FIELDS.AVAILABILITY_ZONES];
+        const defaultAZNameList: string[] = AwsDefaults.defaulAvailabilityZoneNameStrategy(azList, nodeProfile);
+        const selectedAzs: { [key: string]: string }[] = [];
+
+        for (const az of defaultAZNameList) {
+            DefaultOrchestrator.initResources<{ [key: string]: string }>({
+                resourceName: AWS_FIELDS.AVAILIABILITY_ZONE_NODE_TYPES,
+                segment: az,
+                dispatch: awsDispatch,
+                errorObject,
+                setErrorObject,
+                //  (a) store all the instance types in a segmented resource under THAT AZ
+                fetcher: () => {
+                    return new CancelablePromise(async (res, rej) => {
```

Comment:
Is there any particular reason to use CancelablePromise?

Reply:
The type of `fetcher` is `() => CancelablePromise<RESOURCE_TYPE[]>;`
`CancelablePromise` implements `Promise`.
async/await can work in this scenario, but the type check will show error like
```
Type '() => Promise<{ [key: string]: string; }[]>' is not assignable to type '() => CancelablePromise<{ [key: string]: string; }[]>'.
  Type 'Promise<{ [key: string]: string; }[]>' is missing the following properties from type 'CancelablePromise<{ [key: string]: string; }[]>': #isResolved, #isRejected, #isCancelled, #cancelHandlers, and 3 more.ts(2322)
```
So I use `CancelablePromise`